lisa/tools/remote_copy.py

```diff
@@ -0,0 +1,132 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+import os
+from pathlib import PurePath
+
+from lisa.executable import Tool
+from lisa.tools.chown import Chown
+from lisa.tools.cp import Cp
+from lisa.tools.ls import Ls
+from lisa.tools.mkdir import Mkdir
+from lisa.tools.rm import Rm
+from lisa.tools.whoami import Whoami
+
+
+class RemoteCopy(Tool):
+    @property
+    def command(self) -> str:
+        return ""
+
+    @property
+    def can_install(self) -> bool:
+        return False
+
+    def _prepare_tmp_copy_folder(
+        self,
+        src: PurePath,
+        recurse: bool = False,
+    ) -> PurePath:
+        # create a temp directory and copy the files to it
+        tmp_dir = PurePath("/tmp") / src.name
+        if recurse:
+            self.node.tools[Cp].copy(src, tmp_dir.parent, sudo=True, recur=True)
+        else:
+            self.node.tools[Mkdir].create_directory(str(tmp_dir), sudo=True)
+            self.node.tools[Cp].copy(src / "*", tmp_dir, sudo=True)
+
+        # change the owner of the temp directory
+        username = self.node.tools[Whoami].get_username()
+        self.node.tools[Chown].change_owner(
+            tmp_dir, user=username, group=username, recurse=True
+        )
+
+        return tmp_dir
+
+    def _prepare_tmp_copy_file(
+        self,
+        src: PurePath,
+    ) -> PurePath:
+        # create a temp directory and copy the src file to it
+        tmp_dir = PurePath("/tmp")
+        self.node.tools[Cp].copy(src, tmp_dir, sudo=True)
+
+        # change the owner of the file
+        tmp_file = tmp_dir / src.name
+        username = self.node.tools[Whoami].get_username()
+        self.node.tools[Chown].change_owner(tmp_file, user=username, group=username)
+
+        return tmp_file
+
+    def _copy_dir(
```

Comment:
The logic of dir covers the logic of file. Merge them together can make the tool is easy to use.

Reply:
My initial though was to do this as well but I think that `recurse` argument is confusing in context of file and the `_prepare_tmp_copy_*` behaves different for file and folder so we need to this check, and that would be another SSH call.

Reply:
`recurse` is understandable, even if it's copying file. The preparing process should more rely on the behavior of cp. I think it can cover both file and dir scenarios.